packages/frontend/editor-ui/src/features/execution/logs/components/ChatMessagesPanel.vue

```diff
@@ -29,138 +34,167 @@ const props = withDefaults(defineProps<Props>(), {
+			initializeChat();
 		}
+	},
+);
 
-		// Get message at current index
-		const selectedMessage = pastMessages[previousMessageIndex.value];
-		chatEventBus.emit('setInputValue', selectedMessage);
-
-		// Refocus and move cursor to end
-		chatEventBus.emit('focusInput');
-	} else if (key === 'ArrowDown') {
-		// Only navigate if we're in history mode
-		if (previousMessageIndex.value === -1) return;
-
-		// Temporarily blur to avoid cursor position issues
-		chatEventBus.emit('blurInput');
-
-		if (previousMessageIndex.value < pastMessages.length - 1) {
-			// Move forward through history (newer messages)
-			previousMessageIndex.value++;
-			const selectedMessage = pastMessages[previousMessageIndex.value];
-			chatEventBus.emit('setInputValue', selectedMessage);
-		} else {
-			// Reached the end - restore original input or clear
-			previousMessageIndex.value = -1;
-			chatEventBus.emit('setInputValue', currentInputBuffer.value);
-			currentInputBuffer.value = '';
+// Watch for workflow ID changes (important for webhook URL)
+watch(
+	() => workflowsStore.workflowId,
+	async (newWorkflowId, oldWorkflowId) => {
+		if (props.isOpen && isWorkflowReadyForChat.value && newWorkflowId !== oldWorkflowId) {
+			// Workflow ID changed and workflow is ready - reinitialize chat with new webhook URL
+			destroyChat();
+			initializeChat();
 		}
+	},
+);
 
-		// Refocus and move cursor to end
-		chatEventBus.emit('focusInput');
-	}
-}
+// Watch for streaming configuration changes
+watch(
+	() => isStreamingEnabled.value,
+	async (newStreaming, oldStreaming) => {
+		if (props.isOpen && isWorkflowReadyForChat.value && chatApp && newStreaming !== oldStreaming) {
+			// Reinitialize chat when streaming configuration changes and workflow is ready
+			destroyChat();
+			initializeChat();
+		}
+	},
+);
 
-function onEscapeKey() {
-	// Only handle escape if we're in history navigation mode
-	if (previousMessageIndex.value === -1) return;
+// Watch for file upload configuration changes
+watch(
+	() => [isFileUploadsAllowed.value, allowedFilesMimeTypes.value],
+	async (newConfig, oldConfig) => {
+		if (
+			props.isOpen &&
+			isWorkflowReadyForChat.value &&
+			chatApp &&
+			JSON.stringify(newConfig) !== JSON.stringify(oldConfig)
+		) {
+			// Reinitialize chat when file upload configuration changes and workflow is ready
+			destroyChat();
+			initializeChat();
+		}
+	},
+);
 
-	// Exit history mode and restore original input
-	previousMessageIndex.value = -1;
-	chatEventBus.emit('setInputValue', currentInputBuffer.value);
-	currentInputBuffer.value = '';
-}
+onMounted(async () => {
+	if (props.isOpen) {
+		// Wait for next tick to ensure DOM is ready
+		await nextTick();
+		void initializeChat();
+	}
+});
 
-async function copySessionId() {
-	await clipboard.copy(props.sessionId);
-	toast.showMessage({
-		title: locale.baseText('generic.copiedToClipboard'),
-		message: '',
-		type: 'success',
-	});
-}
+onUnmounted(() => {
+	destroyChat();
```

Comment:
Not sure this should be the case either. If I switch to executions tab and go back, the chat history should be there.